lib/events.js

```diff
@@ -385,16 +392,16 @@ EventEmitter.prototype.removeAllListeners =
         this.removeListener(type, listeners);
       } else if (listeners !== undefined) {
         // LIFO order
-        for (i = listeners.length - 1; i >= 0; i--) {
+        for (i = listeners.length - 1; i >= 0; --i) {
           this.removeListener(type, listeners[i]);
         }
       }
 
       return this;
     };
 
-EventEmitter.prototype.listeners = function listeners(type) {
-  const events = this._events;
+function _listeners(target, type, unwrap) {
```

Comment:
I wonder if it would make sense to turn this into a closure where `unwrap` can be constant-folded, like:

```js
function makeListeners(unwrap) {
  return function listeners(type) {
    // Code of _listeners with `this` as `target`
  }
}

EventEmitter.prototype.listeners = makeListeners(true);
EventEmitter.prototype.rawListeners = makeListeners(false);
```

Reply:
This was just to mimic how we do it elsewhere in this file, namely `addListener` and `prependListener` but yeah, this could also work. I suppose the suggested version does close over one variable whereas what I'm guessing happens with the current implementation is that V8 inlines `_listeners`?

No clue what's preferable. Maybe just stick with consistency for now?

Reply:
If there is no perf difference, then yes, just stick with this :)